File: gluecodium/src/test/resources/smoke/visibility_attribute/output/android/com/example/smoke/PublicStructWithInternalConstructors.java
/*

 *
 */

package com.example.smoke;

import android.support.annotation.NonNull;

public final class PublicStructWithInternalConstructors {
    public int someVar;


    PublicStructWithInternalConstructors() {
        PublicStructWithInternalConstructors _other = make();
        this.someVar = _other.someVar;
    }

    PublicStructWithInternalConstructors(final int someVar) {
        this.someVar = someVar;
    }

    private static native PublicStructWithInternalConstructors make();


}

File: gluecodium/src/test/resources/smoke/visibility_attribute/output/android-kotlin/com/example/smoke/PublicStructWithInternalConstructors.kt
/*

 *
 */

@file:JvmName("PublicStructWithInternalConstructorsExtensions")

package com.example.smoke


class PublicStructWithInternalConstructors {
    @JvmField var someVar: Int





   internal constructor() {
        val _other = make()
        this.someVar = _other.someVar
    }



    internal constructor(someVar: Int) {
        this.someVar = someVar
    }





    companion object {

        @JvmStatic @JvmSynthetic @JvmName("make") internal external fun make() : PublicStructWithInternalConstructors
    }
}

